src/resources/Validator.js

```diff
@@ -194,7 +195,20 @@ class Validator {
     if (!resource) {
       throw new Error(`Resource at ${resourceId} not found for dereferencing`);
     }
-    return resource.components || [];
+    const dataTableComponents = (component.fetch.components || [])
```

Comment:
This looks good, although when I look at this a year from now I will ask "what is this?" Can we add a //coment as to why we're filtering this and what this accomplishes? Thanks